src/assignments/files-match-result.md

```diff
@@ -0,0 +1,441 @@
+# Files, Match and Result
+
+In this exercise you will complete a number of steps to learn about
+Error Handling. The final result will be a url parser that reads lines
+from a text file and can distinguish the content between URLs and
+non-urls.
+
+## In this exercise, you will learn how to
+
+-   handle occurring `Result`-types with `match` for basic error
+    handling.
+
+-   when to use the `.unwrap()` method.
+
+-   propagate an error with the `?` operator
+
+-   return the `Option`-type.
+
+-   do some elementary file processing (opening, reading to buffer,
+    counting, reading line by line).
+
+-   navigate the Rust `stdlib` documentation
+
+-   add external dependencies to your project
+
+## Task
+
+✅ Clone the teaching material repository at
+[github.com/ferrous-systems/teaching-material](https://github.com/ferrous-systems/teaching-material).
+
+[todo!] add correct location once it's definite.
+✅ Fix the runtime error by correcting the file path. 
+
+✅ Manually unwrap the `Result` type that is returned from the
+    File::open() with a match statement, so that the .unwrap() can be
+    deleted.
+
+✅ Move this manual unwrap to it’s own function.
+
+✅ Read the content of the file to a buffer and count the lines in a
+    for loop.
+
+✅ Filter out empty lines and print the non-empty ones.
+
+✅ Write a function that parses each line and returns Some(url) if the line is a URL, and `None` if it is not. Use the Url crate [Url
+    Type](https://docs.rs/url/2.1.1/url/)
+
+## Knowledge
+
+### Option and Result
+
+Both `Option` and `Result` are similar in a way. Both have two
+variants, and depending on what those variants are, the program may
+continue in a different way.
+
+The `Option` Type can have the variant `Some(<some other type>)` or
```

Comment:
Uses `Some(<some other type>)` for Option and `Ok(t)` when talking about Result. These should use the same syntax, as they are the same kind of thing.

Suggest using Rust syntax, so:


> The `Option` type can have the variant `Some(T)`, where `T` is a type parameter that means *some type should go here, we'll decide which one later*.